k8s/oidc-vault/README.md

```diff
@@ -0,0 +1,435 @@
+```
+
+## Configure an A Record for the OIDC Discovery Document Endpoint
+
+Using your preferred DNS tool, put the `MY_DISCOVERY_DOMAIN` domain and the `spire-oidc` external IP address in a new DNS A record. The A record should take the following form:
+
+```console
+MY_DISCOVERY_DOMAIN          A        EXTERNAL-IP for spire-oidc service
+```
+
+For example:
+```console
+oidc-discovery.example.org   A        93.184.216.34
+```
+
+Note: Do not use the `oidc-discovery.example.org` domain or IP address shown above.
+
+## Verify the DNS A Record
+
+As with any change to DNS, it will take minutes or hours for the new A record to propagate to DNS servers. This tutorial will not work until the A record is propagated. Negative DNS query results will be cached, causing headaches. So, to be safe, wait an hour or so to test the DNS change after creating the A record.
+
+1. Use `nslookup` to display the DNS information for the domain you configured in the A record:
+
+   ```console
+   $ nslookup oidc-discovery.example.org
+   Server:        203.0.113.0
+   Address:	      203.0.113.0#53
+
+   Non-authoritative answer:
+   Name:	oidc-discovery.example.org
+   Address: 93.184.216.34
+   ```
+
+   The `Address:` field at the bottom should correspond to the IP address in the A record.
+
+2. In your browser, navigate to `https://MY_DISCOVERY_DOMAIN/.well-known/openid-configuration`. You should see a JSON output similar to the following:
+
+   ```json
+   {
+     "issuer": "https://oidc-discovery.example.org",
+     "jwks_uri": "https://oidc-discovery.example.org/keys",
+     "authorization_endpoint": "",
+     "response_types_supported": [
+       "id_token"
+     ],
+     "subject_types_supported": [],
+     "id_token_signing_alg_values_supported": [
+       "RS256",
+       "ES256",
+       "ES384"
+     ]
+   }
+   ```
+
+
+# Part 3: Configure Vault server
+
+After configuring an A Record for the OIDC discovery document endpoint, continue with configuring the Vault server.
+
+## Create the config file and run the Vault server
+
+First, let's create a simple config file for our local Vault server.
+
+1. Create a file on your local computer called `config.hcl` (you can find the file [here](./vault/config.hcl)). Here we define that the vault server will listen on our local interface 127.0.0.1 port 8200. We'll disable TLS for simplicity (don't do this on production) and set a default `file` backend 
+
+   ``` console
+   listener "tcp" {
+      address     = "127.0.0.1:8200",
+      tls_disable = 1
+   }
+
+   storage "file" {
+      path = "vault-storage"
+   }
+   ```
+
+2. In the same directory, run:
+
+   ```console
+   $ vault server -config config.hcl
+   ```
+
+3. Check that your Vault server has started correctly and has no errors, you may see some warnings.
+
+   ```console
+   ==> Vault server started! Log data will stream in below:
+   ```
+
+## Initialize and unseal the vault
```

Comment:
Oops, the sentence I added was a joke meant for you. The title "Unseal the Vault" reminded me of Indiana Jones. It would be fun to leave it in but would probably confuse a lot of people.

Reply:
Ha! I was about to ask you about this and thought it would be fun to leave it there 😃 but yeah you're right, it would confuse people. I'll delete this and save the Indiana Jones stories for a future SPIRE-TempleOfDoom integration tutorial